mocket/utils.py

```diff
@@ -38,34 +46,34 @@ def hexload(string):
     return encode_to_bytes(binascii.unhexlify(string_no_spaces))
 
 
-def get_mocketize(wrapper_):
+def get_mocketize(wrapper_: Callable) -> Callable:
     import decorator
 
-    if decorator.__version__ < "5":  # pragma: no cover
+    if decorator.__version__ < "5":  # type: ignore[attr-defined] # pragma: no cover
         return decorator.decorator(wrapper_)
-    return decorator.decorator(wrapper_, kwsyntax=True)
+    return decorator.decorator(wrapper_, kwsyntax=True)  # type: ignore[call-arg] # kwsyntax
 
 
 class MocketMode:
-    __shared_state = {}
-    STRICT = None
-    STRICT_ALLOWED = None
+    __shared_state: ClassVar[dict[str, Any]] = {}
+    STRICT: ClassVar = None
+    STRICT_ALLOWED: ClassVar = None
 
-    def __init__(self):
+    def __init__(self) -> None:
         self.__dict__ = self.__shared_state
 
-    def is_allowed(self, location: Union[str, Tuple[str, int]]) -> bool:
+    def is_allowed(self, location: str | tuple[str, int]) -> bool:
         """
         Checks if (`host`, `port`) or at least `host`
         are allowed locationsto perform real `socket` calls
         """
         if not self.STRICT:
             return True
-        host, _ = location
+        host, _ = location  # type: ignore[misc] # can't unpack a string
```

Comment:
Is this a bug? Should location only accept the `tuple[str, int]`?

Reply:
It was a bug, but could not happen since today it's only used as:
```python
        if not MocketMode().is_allowed((self._host, self._port)):
            MocketMode.raise_not_allowed()
```

Reply:
I've just fixed it and released a new minor version.